controllers/mattermost/mattermost/mattermost.go

```diff
@@ -350,19 +368,23 @@ func (r *MattermostReconciler) checkUpdateJob(
 	jobNamespace string,
 	baseDeployment *appsv1.Deployment,
 	reqLogger logr.Logger,
-) (*batchv1.Job, error) {
+) (*batchv1.Job, reconcileStatus, error) {
+	recStatus := reconcileStatus{
+		ResourcesReady: true,
+	}
 	reqLogger.Info(fmt.Sprintf("Running Mattermost update image job check for image %s", mmv1beta.GetMattermostAppContainerFromDeployment(baseDeployment).Image))
 	job, err := r.Resources.FetchMattermostUpdateJob(jobNamespace)
 	if err != nil {
 		if k8sErrors.IsNotFound(err) {
 			reqLogger.Info("Launching update image job")
 			if err = r.Resources.LaunchMattermostUpdateJob(mattermost, jobNamespace, baseDeployment, reqLogger); err != nil {
-				return nil, errors.Wrap(err, "Launching update image job failed")
+				return nil, reconcileStatus{}, errors.Wrap(err, "Launching update image job failed")
 			}
-			return nil, errors.New("Began update image job")
+			recStatus.ResourcesReady = false
+			return nil, recStatus, errors.Wrap(err, "failed to restart update job")
```

Comment:
Here we actually managed to restart the job successfully, so we should not return an error, just that resource is not ready.

```suggestion
			return nil, recStatus, nil
```